test/framework/src/main/java/org/elasticsearch/test/rest/ESRestTestCase.java

```diff
@@ -292,7 +292,10 @@ protected HttpHost buildHttpHost(String host, int port) {
     @After
     public final void cleanUpCluster() throws Exception {
         if (preserveClusterUponCompletion() == false) {
-            ensureNoInitializingShards();
+            if (nodeVersions.stream().filter(version -> version.before(Version.V_6_2_0)).findAny().isPresent() == false) {
```

Comment:
I think we can simplify `.filter().findAny().isPresent() == false` to just `.noneMatch()`